contracts/contracts/ProviderRegistry.sol

```diff
@@ -194,54 +208,57 @@ contract ProviderRegistry is
         require(successFee, "fee recipient transfer failed");
     }
 
-    /**
-     * @dev Withdraw funds to the bidder.
-     * @param bidder The address of the bidder.
-     */
-    function withdrawBidderAmount(address bidder) external nonReentrant {
-        require(bidderAmount[bidder] > 0, "Bidder Amount is zero");
-
-        bidderAmount[bidder] = 0;
-
-        (bool success, ) = bidder.call{value: bidderAmount[bidder]}("");
-        require(success, "Couldn't transfer to bidder");
+    /// @dev Requests withdrawal of the staked amount.
+    function unstake() external {
+        require(providerStakes[msg.sender] != 0, "No stake to withdraw");
+        require(withdrawalRequests[msg.sender] == 0, "Unstake request exists");
+        withdrawalRequests[msg.sender] = block.timestamp;
+        emit Unstake(msg.sender, block.timestamp);
     }
 
-    /**
-     * @dev Withdraw staked amount for the provider.
-     * @param provider The address of the provider.
-     */
-    function withdrawStakedAmount(
-        address payable provider
-    ) external nonReentrant {
-        require(msg.sender == provider, "Only provider can unstake");
-        uint256 stake = providerStakes[provider];
-        providerStakes[provider] = 0;
-        require(stake > 0, "Provider Staked Amount is zero");
-        require(
-            preConfirmationsContract != address(0),
-            "preconf contract not set"
-        );
+    /// @dev Completes the withdrawal of the staked amount.
+    function withdraw() external nonReentrant {
+        require(withdrawalRequests[msg.sender] != 0, "No unstake request");
+        require(block.timestamp >= withdrawalRequests[msg.sender] + withdrawalPeriod, "Period has not passed");
```

Comment:
I believe the mev-commit chain's `block.timestamp` is in milliseconds? cc @ckartik 

I'd suggest using blocks for _period of time_ parameters as is done with the validator registry contracts

Reply:
Good catch with milliseconds! It’s safe to use block.timestamp here. Typically, block.number is recommended when there’s a potential risk of a miner manipulating the timestamp. However, in this case, since the waiting time is 24 hours (not milliseconds), such manipulation is not a concern.

Reply:
Yes the withdraw period is being set to `24 * 3600`, and the comments specify `withdrawalPeriod` is in seconds, but the `block.timestamp` of our chain is in milliseconds. So the effective withdraw period enforced on this line will be 1000x faster than what's specified. Unless I'm misunderstanding something

Reply:
You 100% right! I was uploading a fix for that...

Reply:
The fact that our chain has changed the unit of `block.timestamp` is why I was suggesting to use number of blocks, but timestamp is fine too as long as we specify correct units